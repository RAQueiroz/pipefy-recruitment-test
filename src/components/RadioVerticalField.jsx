import React from "react";
import styled from "styled-components";
import StyledLabel from "./StyledLabel";
import { InlineError } from "./StyledElements";

const StyledRadio= styled.div`
  border: solid 1px ${({ checked }) => (checked ? "#006699" : "#CCC")};
  width: 20px;
  height: 20px;
  margin-right: 8px;
  border-radius: 50%;
  display: inline-block;
  &:before {
    content: "âœ”";
    position: relative;
    top: 0px;
    left: 3px;
    color: ${({ checked }) => (checked ? "#006699" : "#FFF")};
  }
`;

const RadioVerticalField = ({
  label,
  id,
  error,
  options,
  value,
  onChange = () => {}
}) => (
  <div id={id}>
    <StyledLabel>{label}</StyledLabel>
    {options.map((item, index) => (
      <label
        data-testid={`radio-${index}`}
        key={item}
        style={{ display: "block", padding: 10, cursor: "pointer" }}
        onClick={() =>
          onChange({
            id,
            value: item
          })
        }
      >
        <StyledRadio checked={value === item}  />
        {item}
      </label>
    ))}
    {error && <InlineError>{error}</InlineError>}
  </div>
);

export default RadioVerticalField;
