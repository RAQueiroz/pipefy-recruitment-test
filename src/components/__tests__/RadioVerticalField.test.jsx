import React from "react";
import RadioVerticalField from "../RadioVerticalField";
import { render, Simulate, wait } from "react-testing-library";
// import "dom-testing-library/extend-expect";

describe(" <RadioVerticalField/>", () => {
  const item = {
    id: "xpto",
    label: "Name",
    value: "",
    options: ["A", "B", "C"]
  };

  const onChangeSpy = jest.fn();
  const { getByText, getByTestId, container, queryByText } = render(
    <RadioVerticalField onChange={onChangeSpy} {...item} />
  );

  it("renders", () => {
    const submitButton = queryByText("Name");
    expect(submitButton).toBeTruthy(); // it exist
  });

  it("render correct number of radios", () => {
    item.options.forEach((item, index) => {
      expect(getByTestId(`radio-${index}`)).toBeTruthy();
    });
  });

  it("match snapshot", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it("on change to be called with the correct value", () => {
    Simulate.click(getByTestId(`radio-1`));
    expect(onChangeSpy).toHaveBeenCalledWith({ id: "xpto", value: "B" });
  });
});
