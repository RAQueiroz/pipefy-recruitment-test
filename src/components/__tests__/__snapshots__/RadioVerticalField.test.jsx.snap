// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <RadioVerticalField/> match snapshot 1`] = `
<div
  id="xpto"
>
  <label
    class="sc-bdVaJa hOggSJ"
  >
    Name
  </label>
  <label
    data-testid="radio-0"
    style="display: block; padding: 10px; cursor: pointer;"
  >
    <div
      class="sc-gzVnrw ihXyvo"
    />
    A
  </label>
  <label
    data-testid="radio-1"
    style="display: block; padding: 10px; cursor: pointer;"
  >
    <div
      class="sc-gzVnrw ihXyvo"
    />
    B
  </label>
  <label
    data-testid="radio-2"
    style="display: block; padding: 10px; cursor: pointer;"
  >
    <div
      class="sc-gzVnrw ihXyvo"
    />
    C
  </label>
</div>
`;
